test/integration/auth/node_test.go

```diff
@@ -664,3 +672,21 @@ func expectAllowed(t *testing.T, f func() error) {
 		t.Errorf("Expected no error, got %v", err)
 	}
 }
+
+// crdFromManifest reads a .json/yaml file and returns the CRD in it.
+func crdFromManifest(filename string) (*apiextensionsv1beta1.CustomResourceDefinition, error) {
+	var crd apiextensionsv1beta1.CustomResourceDefinition
+	data, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return nil, err
+	}
+
+	json, err := utilyaml.ToJSON(data)
```

Comment:
Question: is the conversion from yaml to json here actually necessary? I though the universal decoder handled all serialization media types automatically.

Reply:
Cleaned up.